### PR TITLE
fixes /devices filters

### PR DESCRIPTION
**Pull request by tomershinhar**
# Description

the /devices endpoint filters was not matching the devices filters listed in query_params.go which caused the endpoint to not except valid query params and also passing invalid one (which is not actually doing anything).
this PR updates the query_params.go file to match the query params accepted by the /devices endpoint and also updates the spec file to better document how to use those query params 

Fixes # (issue)
[THEEDGE-2717](https://issues.redhat.com/browse/THEEDGE-2717)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by AaronH88 — APPROVED** [review]
/lgtm

File: pkg/routes/query_params.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func initalizeQueryParamsArray() map[string][]string {
 	if len(m) == 0 {
 		m = make(map[string][]string)
 		m["device-groups"] = []string{"limit", "offset", "name", "created_at", "updated_at", "sort_by"}
-		m["devices"] = []string{"per_page", "page", "sort_by", "order_how", "hostname_or_id"}
+		m["devices"] = []string{"per_page", "page", "order_how", "hostname_or_id", "order_by"}
```
Comment (ldjebran) [inline pkg/routes/query_params.go:21]: I see sort_by and not order_by  
https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/common/filters.go#L120

File: cmd/spec/path.yaml [hunk 2]
```diff
@@ -629,12 +629,12 @@ paths:
             type: integer
         - name: order_by
```
Comment (ldjebran) [inline cmd/spec/path.yaml:630]: may be this line should be replaced by sort_by as per https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/common/filters.go#L120

**Review by ldjebran — CHANGES_REQUESTED** [review]
I see sort_by and not order_by here
https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/common/filters.go#L120

File: pkg/routes/query_params.go (hunk 1 above)
Comment (tomershinhar) [reply pkg/routes/query_params.go:21]: for the /devices endpoint the filters are taken from host-inventory https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/devices.go#L252
so while sort_by was allowed it did not do anything

**Review by ldjebran — APPROVED** [review]

**Pull request merged**